core/src/main/kotlin/builder/kord/KordBuilder.kt

```diff
@@ -159,6 +157,23 @@ public class KordBuilder(public val token: String) {
         this.handlerBuilder = handlerBuilder
     }
 
+    /**
+     * Enables stack trace recovery on the currently defined [RequestHandler].
+     *
+     * @throws IllegalStateException if the [RequestHandler] is not a [KtorRequestHandler]
+     *
+     * @see StackTraceRecoveringKtorRequestHandler
+     * @see withStackTraceRecovery
+     */
+    public fun enableStackTraceRecovery() {
+        val parentBuilder = handlerBuilder
+        requestHandler {
+            val ktorRequestHandler = parentBuilder(it) as? KtorRequestHandler
+                ?: error("Stack trace recovery only works with KtorRequestHandlers")
+            ktorRequestHandler.withStackTraceRecovery()
+        }
+    }
```

Comment:
I think this should set some sort of flag to use recovery that is then used at build time to replace the handler instead of replacing the `handlerBuilder`, the following would not work as intended because `requestHandler { }` replaces the recovering handler:
```kotlin
val kord = Kord("") {
    enableStackTraceRecovery()
    requestHandler { KtorRequestHandler(...) }
}
```
It could also just be a Boolean property in that case:
```kotlin
val kord = Kord("") {
    stackTraceRecovery = true
}
```

Apart from that: you only added it to `KordBuilder` and not `KordRestOnlyBuilder`.

Reply:
With that flag, there would be no validation whether it actually does something or not, since it only works with the KtorRequestHandler

Reply:
If you put the logic inside `requestHandler { }` in `enableStackTraceRecovery` into the build function behind an if check there would.

Reply:
requestHandler currently doesn't have a builder it's just a function returning a RequestHandler, so that would require changing a lot of things

Reply:
Let me clarify what I mean:
```kotlin
class KordBuilder {

    var recoverStacktraces = false

    // ...

    fun build(): Kord {
        // ...
        val resources = ...
        var handler = handlerBuilder(resources)
        if (recoverStacktraces) {
            require(handler is KtorRequestHandler)
            handler = handler.withStackTraceRecovery
        }
        // ...
    }
}
```